### PR TITLE
Modal comp

### PR DESCRIPTION
**Pull request by isandeepkadam**

### CODE REVIEW

File: client/src/components/Buttons.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import type { Component } from "solid-js"
+
+export const PrimaryButton: Component<{
```
Comment (shubham-cpp) [inline client/src/components/Buttons.tsx:3]: **PrimaryButton** and **SecondaryButton**?.

Why not `<Button variant="secondary" />`
Default variant would be **primary**.
Also the default `type` of button should be `type="button"`. I dont know why the default type="submit".
[Mdn Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type)

File: client/src/components/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const [show, setShow] = createSignal(false)
+
+  return (
+    <Show when={show()} fallback={<PrimaryButton label={props.buttonTitle || 'Undefined'} onClick={() => setShow(true)}/>}>
+    <div class="modal-wrapper bg-slate-900 bg-opacity-70 z-50 fixed h-screen w-screen top-0 left-0 flex justify-center items-center">
+      <div role="alertdialog" aria-modal="true" aria-labelledby={`modal-${props.title}`} 
+            class="modal align-super bg-white w-3/4 min-h-0 max-h-3/4 rounded-sm bg-opacity-100 shadow-lg overflow-y-auto p-10">
+        <div class="modal-header flex align-super justify-between pb-2">
+          <h1 class="text-lg">{props.title}</h1>
+          <button onClick={() => setShow(false)}>❌</button>  {/** using Emoji for now */}
```
Comment (shubham-cpp) [inline client/src/components/Modal.tsx:23]: Use Astro icons, instead of emoji

File: client/src/components/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  actionOnClick?: () => void
+  children?: JSXElement
+}
+
+const Modal: Component<ModalProps> = props => {
+  const [show, setShow] = createSignal(false)
+
+  return (
+    <Show when={show()} fallback={<PrimaryButton label={props.buttonTitle || 'Undefined'} onClick={() => setShow(true)}/>}>
+    <div class="modal-wrapper bg-slate-900 bg-opacity-70 z-50 fixed h-screen w-screen top-0 left-0 flex justify-center items-center">
```
Comment (shubham-cpp) [inline client/src/components/Modal.tsx:18]: Why not native [Dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog). Its built-in with html. Also seems to have good support [check here](https://caniuse.com/?search=dialog)

**Comment by isandeepkadam** [comment]
@shubham-cpp  bruhh

File: client/src/components/Inputs/SelectInput.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Component, For } from "solid-js";
+
+interface SelectInuputProps{
+  label?: string
+  options: string[]
```
Comment (shubham-cpp) [inline client/src/components/Inputs/SelectInput.tsx:5]: make `options`, a `string[]` or array of `Option[]`
```typescript
type Option {
  value: string;
  label: string;
}
``` 

### Why?
Consider following scenario.
I want to select project from list, now we're stored project names as INT or ids,etc but we want the label of option to show Integration, Oracle Database or whatever actual name of project is.

File: client/src/components/Inputs/SelectInput.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  return (
+  <label aria-label={`${props.label}-select-label`}>{props.label}
+    <select name="choice" class="bg-gray-200 text-gray-700 border border-gray-300 rounded-lg w-64 py-2 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500 p-4 mx-2" 
+            onChange={handleOptionChange}
+    >
+      <For each={props.options}>
+        {(option, index) => (
+          <option value={option} id={`${index}-${option}`}>
```
Comment (shubham-cpp) [inline client/src/components/Inputs/SelectInput.tsx:23]: if we're not using `id` we can remove it

File: client/tsconfig.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
   "extends": "astro/tsconfigs/strictest",
   "compilerOptions": {
     "jsx": "preserve",
-    "jsxImportSource": "solid-js"
+    "jsxImportSource": "solid-js",
+    "allowImportingTsExtensions": true,
```
Comment (shubham-cpp) [inline client/tsconfig.json:6]: could you mention reasons behind this?

**Review by shubham-cpp — CHANGES_REQUESTED** [review]
Modal looks good

**Comment by isandeepkadam** [comment]
@shubham-cpp changes done

**Pull request merged**